### PR TITLE
highlights(vala): Coloring fixes

### PR DESCRIPTION
**Pull request by matbme**
Simplified grammar syntax, update highlights to reflect changes

### CODE REVIEW

File: queries/vala/highlights.scm [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-    (camel_cased_identifier) @namespace
-    (identifier) @variable
+    (identifier) @namespace
+    (_)
   ]
   right: [
-    (identifier) @parameter
-    (camel_cased_identifier) @type
-    (uppercased_identifier) @constant
+    ((identifier) @parameter (#match? @parameter "^[a-z]"))
```
Comment (theHamsta) [inline queries/vala/highlights.scm:7]: every namespaced_indentifier is a `@parameter` ?

File: queries/vala/highlights.scm [hunk 2]
```diff
@@ -247,12 +236,14 @@
...
 (new_instance
-  ".new" @keyword
+  "new" @keyword
 )
 
 ; GObject construct
 
-"construct" @constructor
+(gobject_construct 
+  "construct" @constructor
```
Comment (theHamsta) [inline queries/vala/highlights.scm:245]: this should probably be a keyword. `@constructor` in other languages are often the functions/types called to construct an object.

**Comment by clason** [comment]
Can you include the bump to the latest version of the Vala parser here and make sure the queries pass?

**Comment by matbme** [comment]
I'm confused as to why the checks are failing. `tree-sitter test` returns no errors

**Comment by clason** [comment]
You probably have already updated the parser locally in your lockfile but forgot to `git add` it?

**Comment by clason** [comment]
You probably have already updated the parser locally in your lockfile but forgot to `git add` it?

**Comment by clason** [comment]
Did you really intend to rewrite `lockfile.json` into a Lua table o.O`

**Comment by clason** [comment]
Don't run the script to update the lockfile; just replace the commit hash manually _just_ for the Vala parser.

**Comment by matbme** [comment]
The update lockfile script generated an empty lockfile and I was ~stupid~ careless enough to just copy the command output into the file. I'll do what you suggested

**Comment by theHamsta** [comment]
I'd recommend to do
```
git fetch upstream              # assuming nvim-treesitter/nvim-treesitter remote is called upstream
git reset upstream/master  # assuming nvim-treesitter/nvim-treesitter remote is called upstream
# re-commit your changes, drop all files the should not be changed, maybe update the SHA of the vala parser
git push --force
```

**Comment by clason** [comment]
I'd just `git rebase -i HEAD~3` (or however many commits are in your branch), drop the last two, add the new commit with the manual change, then `git push -f`.

**Comment by matbme** [comment]
Crisis averted :0

**Review by theHamsta — APPROVED** [review]

**Pull request merged**